src/tensor_parallel/tensor_parallel.py

```diff
@@ -130,6 +137,8 @@ def forward(self, *args, **kwargs):
 
     def state_dict(self, *args, **kwargs):
         state_dict = super().state_dict(*args, **kwargs)
```

Comment:
Does this state dict break the relation with load_state_dict?

As in, can I still model1.load_state_dict(model2.state_dict())

If not, perhaps it's better to implement a separate function from state_dict (e.g. consolidated_state_dict?)

Reply:
Changed it a little. Now it works like this:
```python
model = <some model>
model_tp = tensor_parallel(model)

# LEGAL
model_tp.load_state_dict(model_tp.state_dict()) # Just a normal state_dict with no shenanigans

# LEGAL
with save_tensor_parallel(model_tp):
    model.load_state_dict(model_tp.state_dict()) # recontsructing what is expected of model.state_dict()

```